.circleci/scripts/binary_checkout.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+set -eux -o pipefail
+
+retry () {
+    $*  || (sleep 1 && $*) || (sleep 2 && $*) || (sleep 4 && $*) || (sleep 8 && $*)
+}
+
+
+# This step runs on multiple executors with different envfile locations
+if [[ "$OSTYPE" == "msys" ]]; then
+  # windows executor (builds and tests)
+  rm -rf /c/w
+  ln -s "${HOME}" /c/w
+  WORK_DIR="/c/w"
+elif [[ -d "/home/circleci/project" ]]; then
+  # machine executor (binary tests)
+  WORK_DIR="${HOME}/project"
+else
+  # macos executor (builds and tests)
+  # docker executor (binary builds)
+  WORK_DIR="${HOME}"
+fi
+
+if [[ "$OSTYPE" == "msys" ]]; then
+  # We need to make the paths as short as possible on Windows
```

Comment:
why?

Reply:
This is from the previous developer. Keeping the logic for now.